packages/vuetify/src/composables/validation.ts

```diff
@@ -56,7 +56,7 @@ export function useValidation (
   const form = useForm()
   const internalErrorMessages = ref<string[]>([])
   const isPristine = ref(true)
-  const isDirty = computed(() => wrapInArray(model.value || []).length > 0)
```

Comment:
old code seems try to convert falsy values to `[]`, however, `0` should  not be treated as falsy in this case. Here is summary of what should happen:

- for `null` & `undefined`, `wrapInArray` should return `[]` which has been implemented in `wrapInArray`
- for `0`, it should be passed as `0` and `wrapInArray` will return `[0]` as expected
- for `''` however, it should be converted to `null` ,  `wrapInArray` will return `[]` which makes `isDirty` as `false`

Therefore, in terms of `isDirty` calculation, only empty string `''` should be specifically addressed